### PR TITLE
Switch argument santization to regex.

### PR DESCRIPTION
**Pull request by jasedit**
This refactors the argument sanitization changes in #1094 to build a single regex statement for replace problematic characters, which is more easily extensible by adding entries into a dictionary as opposed to stringing `replace` calls together. I'm happy to refactor/adjust as requested, since this is a bit more complicated than the original version.

### CODE REVIEW

File: errbot/__init__.py [hunk 1]
```diff
@@ -346,7 +346,10 @@ def wrapper(self, msg, args):
                 # Same happens with quotations marks, which are required for parsing
                 # complex strings in arguments
                 try:
-                    args = shlex.split(args.replace('—', '--').replace('“', '"').replace('”', '"'))
+                    #Map of strings to replace with sanitized versions
+                    sanitizers = {'—': '--', '“': '"', '”': '"'}
```
Comment (zoni) [inline errbot/__init__.py]: Can you place this list near the top of the file, like: `ARG_BOTCMD_CHARACTER_REPLACEMENTS = {...}`? That will make it easier to locate next time anyone wants to update this list.

(Don't forget to move the comment block above as well)

**Comment by gbin** [comment]
It looks like it broke some (see CI) ;)

**Comment by robojase** [comment]
Finally got some time to fix up the CI, so this is back on track for consideration.

File: errbot/__init__.py [hunk 2]
```diff
@@ -340,13 +348,12 @@ def decorator(func):
...
 
-                # Some clients automatically convert consecutive dashes into a fancy
-                # hyphen, which breaks long-form arguments. Undo this conversion to
-                # provide a better user experience.
-                # Same happens with quotations marks, which are required for parsing
-                # complex strings in arguments
+                # Attempt to sanitize arguments of bad characters
                 try:
-                    args = shlex.split(args.replace('—', '--').replace('“', '"').replace('”', '"'))
+                    print(args)
```
Comment (zoni) [inline errbot/__init__.py]: This looks like a leftover debug statement.

**Comment by robojase** [comment]
Sorry - had removed the DBO in one commit, then accidentally re-added it when cleaning up some of the coding style checks from CI. Should be fixed now.

**Comment by zoni** [comment]
Thanks!

**Pull request merged**